ycmd.el

```diff
@@ -811,7 +813,20 @@ process with `delete-process'."
   "Sends an unspecified message to the server.
 
 This is simply for keepalive functionality."
-  (ycmd--request "/healthy" '() :type "GET"))
+  (ycmd--request "/healthy" nil :type "GET"))
+
+(defun ycmd--server-ready? (include-subserver)
+  "Send request for server ready state."
+  (let ((file-type (car-safe (ycmd-major-mode-to-file-types
+                              major-mode)))
+        (request-message-level -1)
```

Comment:
Here we have to suppress all errors. We only care for ready or not. When the server did not finished starting we get connection errors (with `curl`) which we don't want to see. With `url-retrieve` we get different errors which we suppress with `ignore-errors`